pkg/phlare/modules.go

```diff
@@ -217,6 +288,14 @@ func (f *Phlare) initServer() (services.Service, error) {
 	// see https://github.com/grafana/phlare/issues/231
 	f.Cfg.Server.DoNotAddDefaultHTTPMiddleware = true
 
+	// Disable timeouts if query-scheduler is enabled since it requires long streaming RPCs.
+	if f.isModuleActive(QueryScheduler) {
+		// todo: however we should have API timeout for queries when running a single binary.
+		f.Cfg.Server.HTTPServerIdleTimeout = 0
+		f.Cfg.Server.HTTPServerWriteTimeout = 0
+		f.Cfg.Server.HTTPServerReadTimeout = 0
```

Comment:
`0` is quite long, maybe we can find some middle ground. How big is the disruption when a timeout happens? 

I am thinking of scenarios like:
- if the connection is really dead on a e.g. TCP level?
- how do certificates ever getting renewed?

Reply:
For context the frontend and querier will enter this loop on reaching timeout:

```
backoffConfig := backoff.Config{
		MinBackoff: 250 * time.Millisecond,
		MaxBackoff: 2 * time.Second,
	}
	backoff := backoff.New(ctx, backoffConfig)
	for backoff.Ongoing() {
		if !attemptLoop() {
			backoff.Wait()
		} else {
			backoff.Reset()
		}
	}
```
So 250ms before retrying.

Reply:
This is how it looks like 

```
level=debug caller=gokit.go:65 caller=http.go:183 traceID=4a5eaf1a36f6d586 msg="POST /schedulerpb.SchedulerForFrontend/FrontendLoop (200) 29.996462956s"
level=debug caller=gokit.go:65 caller=http.go:183 traceID=1e01b3ce743005ec msg="POST /schedulerpb.SchedulerForFrontend/FrontendLoop (200) 30.001080655s"
level=debug caller=gokit.go:65 caller=http.go:183 traceID=1b5d7beb9d0fdb9d msg="POST /schedulerpb.SchedulerForFrontend/FrontendLoop (200) 29.996441541s"
level=error caller=scheduler_processor.go:128 ts=2023-02-06T10:52:44.898988Z component=querier-worker msg="error processing requests from scheduler" err="rpc error: code = Internal desc = stream terminated by RST_STREAM with error code: INTERNAL_ERROR" addr=192.168.1.194:4100
level=error caller=scheduler_processor.go:128 ts=2023-02-06T10:52:44.899003Z component=querier-worker msg="error processing requests from scheduler" err="rpc error: code = Internal desc = stream terminated by RST_STREAM with error code: INTERNAL_ERROR" addr=192.168.1.194:4100
level=error caller=scheduler_processor.go:128 ts=2023-02-06T10:52:44.90318Z component=querier-worker msg="error processing requests from scheduler" err="rpc error: code = Internal desc = stream terminated by RST_STREAM with error code: INTERNAL_ERROR" addr=192.168.1.194:4100
level=error caller=scheduler_processor.go:128 ts=2023-02-06T10:52:44.903196Z component=querier-worker msg="error processing requests from scheduler" err="rpc error: code = Internal desc = stream terminated by RST_STREAM with error code: INTERNAL_ERROR" addr=192.168.1.194:4100
```